File: initialize_my_name.py
def initialize_names(name):
    names = name.split(" ")
    new_name = []
    for i in range(len(names)):
        if names[i] != names[0] and names[i] != names[-1]:
            char = f"{names[i][0]}."
            new_name.append(char)
        else:
            new_name.append(names[i])
            
    return " ".join(new_name)

File: initializeMyName.js
function initializeNames(name){
    let nameArr = name.split(' ')
    let newName = []

    for(let i = 0; i < nameArr.length; i++){
        if(nameArr[i] !== nameArr[0] || nameArr[i] !== nameArr[nameArr.length - 1]){
            let char = `${nameArr[i][0]}.`
            newName.push(char)
        }else{
            newName.push(nameArr[i])
        }
    }
    return newName.join(' ')
}